### PR TITLE
Feat/ SHA256 function transpilation to Spark

### PR DESCRIPTION
**Pull request by fireis**
Hey there, my first PR here so sorry for the eventual noob mistakes!

This PR adds Spark transpilation for [SHA256 BigQuery function](https://cloud.google.com/bigquery/docs/reference/standard-sql/hash_functions#sha256) to Spark.  The SHA256 function on BQ uses the SHA-2 algorithm with the 256 length, hence the use of Spark's [SHA2](https://spark.apache.org/docs/3.1.2/api/python/reference/api/pyspark.sql.functions.sha2.html) with fixed length.

### CODE REVIEW

File: sqlglot/dialects/spark2.py [hunk 1]
```diff
@@ -218,6 +218,7 @@ class Generator(Hive.Generator):
                 e.args["replacement"],
                 e.args.get("position"),
             ),
+            exp.SHA256: lambda self, e: f"SHA2({self.sql(e, 'this')}, 256)",
```
Comment (tobymao) [inline sqlglot/dialects/spark2.py:221]: self.func("SHA2", e.this, 256)

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -4729,6 +4729,11 @@ class SHA2(Func):
     arg_types = {"this": True, "length": False}
 
 
+class SHA256(Func):
```
[the remark this answers is not in the document]
Comment (fireis) [reply sqlglot/expressions.py]: I've deleted it and moved to the BigQuery side, as the SHA256 seems to be specific to its dialect

File: sqlglot/dialects/bigquery.py [hunk 3]
```diff
@@ -301,6 +301,8 @@ class Parser(parser.Parser):
                 if re.compile(str(seq_get(args, 1))).groups == 1
                 else None,
             ),
+            "SHA256": lambda args: exp.SHA2(this=seq_get(args, 0), length="256"),
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:304]:
```suggestion
            "SHA256": lambda args: exp.SHA2(this=seq_get(args, 0), length=exp.Literal.number(256)),
```

File: sqlglot/dialects/bigquery.py [hunk 4]
```diff
@@ -301,6 +301,8 @@ class Parser(parser.Parser):
...
+            "SHA512": lambda args: exp.SHA2(this=seq_get(args, 0), length="512"),
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:305]:
```suggestion
            "SHA512": lambda args: exp.SHA2(this=seq_get(args, 0), length=exp.Literal.number("512")),
```

File: sqlglot/dialects/bigquery.py [hunk 5]
```diff
@@ -301,6 +301,8 @@ class Parser(parser.Parser):
                 if re.compile(str(seq_get(args, 1))).groups == 1
                 else None,
             ),
+            "SHA256": lambda args: exp.SHA2(this=seq_get(args, 0), length=exp.Literal.number(256)),
+            "SHA512": lambda args: exp.SHA2(this=seq_get(args, 0), length=exp.Literal.number("512")),
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:305]:
```suggestion
            "SHA512": lambda args: exp.SHA2(this=seq_get(args, 0), length=exp.Literal.number(512)),
```

File: sqlglot/dialects/bigquery.py [hunk 6]
```diff
@@ -431,6 +433,9 @@ class Generator(generator.Generator):
                     _alias_ordered_group,
                 ]
             ),
+            exp.SHA2: lambda self, e: self.func(
+                f"SHA256" if e.args.get("length") == "256" else "SHA512", f"{self.sql(e, 'this')}"
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:437]:
```suggestion
                f"SHA256" if e.args.get("length") == "256" else "SHA512", e.this
```

File: sqlglot/dialects/bigquery.py [hunk 7]
```diff
@@ -431,6 +433,9 @@ class Generator(generator.Generator):
                     _alias_ordered_group,
                 ]
             ),
+            exp.SHA2: lambda self, e: self.func(
+                f"SHA256" if e.args.get("length") == "256" else "SHA512", e.this
```
Comment (tobymao) [inline sqlglot/dialects/bigquery.py:437]:
```suggestion
                f"SHA256" if e.text("length") == "256" else "SHA512", e.this
```

**Pull request merged**